### PR TITLE
Refactor ABM code: Rename EPI to MIO and migration to mobility and World to Model

### PR DESCRIPTION
**Pull request by xsaschako**
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

- changed "EPI" to "MIO"
- changed all appearances of "migration" or similar to "mobility" or similar
- renamed "ABM::World" to "ABM::Model"

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [x] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development)

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

### CODE REVIEW

File: pycode/memilio-simulation/memilio/simulation/osecir.cpp [hunk 1]
```diff
@@ -285,17 +286,16 @@ PYBIND11_MODULE(_simulation_osecir, m)
...
+           mio::Graph<mio::osecir::Model<double>, mio::MobilityParameters<double>>& params_graph,
            size_t contact_locations_size) {
-            auto migrating_comp = {mio::osecir::InfectionState::Susceptible, mio::osecir::InfectionState::Exposed,
-                                   mio::osecir::InfectionState::InfectedNoSymptoms,
-                                   mio::osecir::InfectionState::InfectedSymptoms,
-                                   mio::osecir::InfectionState::Recovered};
-            auto weights        = std::vector<ScalarType>{0., 0., 1.0, 1.0, 0.33, 0., 0.};
-            auto result = mio::set_edges<ContactLocation, mio::osecir::Model<double>, mio::MigrationParameters<double>,
-                                         mio::MigrationCoefficientGroup, mio::osecir::InfectionState,
+            auto moving_comp = {mio::osecir::InfectionState::Susceptible, mio::osecir::InfectionState::Exposed,
```
Comment (jubicker) [inline pycode/memilio-simulation/memilio/simulation/osecir.cpp:291]:
```suggestion
            auto mobile_comp = {mio::osecir::InfectionState::Susceptible, mio::osecir::InfectionState::Exposed,
```

File: pycode/memilio-simulation/memilio/simulation/osecir.cpp [hunk 2]
```diff
@@ -285,17 +286,16 @@ PYBIND11_MODULE(_simulation_osecir, m)
...
+                                mio::osecir::InfectionState::InfectedNoSymptoms,
+                                mio::osecir::InfectionState::InfectedSymptoms, mio::osecir::InfectionState::Recovered};
+            auto weights     = std::vector<ScalarType>{0., 0., 1.0, 1.0, 0.33, 0., 0.};
+            auto result = mio::set_edges<ContactLocation, mio::osecir::Model<double>, mio::MobilityParameters<double>,
+                                         mio::MobilityCoefficientGroup, mio::osecir::InfectionState,
                                          decltype(mio::read_mobility_plain)>(
-                data_dir, params_graph, migrating_comp, contact_locations_size, mio::read_mobility_plain, weights);
+                data_dir, params_graph, moving_comp, contact_locations_size, mio::read_mobility_plain, weights);
```
Comment (jubicker) [inline pycode/memilio-simulation/memilio/simulation/osecir.cpp:298]:
```suggestion
                data_dir, params_graph, mobile_comp, contact_locations_size, mio::read_mobility_plain, weights);
```

File: pycode/memilio-simulation/memilio/simulation/osecirvvs.cpp [hunk 3]
```diff
@@ -326,25 +327,25 @@ PYBIND11_MODULE(_simulation_osecirvvs, m)
...
-                                   mio::osecirvvs::InfectionState::InfectedSymptomsPartialImmunity,
-                                   mio::osecirvvs::InfectionState::ExposedImprovedImmunity,
-                                   mio::osecirvvs::InfectionState::InfectedNoSymptomsImprovedImmunity,
-                                   mio::osecirvvs::InfectionState::InfectedSymptomsImprovedImmunity};
-            auto weights        = std::vector<ScalarType>{0., 0., 1.0, 1.0, 0.33, 0., 0.};
-            auto result         = mio::set_edges<ContactLocation, mio::osecirvvs::Model<double>,
-                                         mio::MigrationParameters<double>, mio::MigrationCoefficientGroup,
-                                         mio::osecirvvs::InfectionState, decltype(mio::read_mobility_plain)>(
-                data_dir, params_graph, migrating_comp, contact_locations_size, mio::read_mobility_plain, weights);
+            auto moving_comp = {mio::osecirvvs::InfectionState::SusceptibleNaive,
```
Comment (jubicker) [inline pycode/memilio-simulation/memilio/simulation/osecirvvs.cpp:332]:
```suggestion
            auto mobile_comp = {mio::osecirvvs::InfectionState::SusceptibleNaive,
```

File: pycode/memilio-simulation/memilio/simulation/osecirvvs.cpp [hunk 4]
```diff
@@ -326,25 +327,25 @@ PYBIND11_MODULE(_simulation_osecirvvs, m)
...
+                                mio::osecirvvs::InfectionState::InfectedNoSymptomsPartialImmunity,
+                                mio::osecirvvs::InfectionState::InfectedSymptomsPartialImmunity,
+                                mio::osecirvvs::InfectionState::ExposedImprovedImmunity,
+                                mio::osecirvvs::InfectionState::InfectedNoSymptomsImprovedImmunity,
+                                mio::osecirvvs::InfectionState::InfectedSymptomsImprovedImmunity};
+            auto weights     = std::vector<ScalarType>{0., 0., 1.0, 1.0, 0.33, 0., 0.};
+            auto result      = mio::set_edges<ContactLocation, mio::osecirvvs::Model<double>,
+                                              mio::MobilityParameters<double>, mio::MobilityCoefficientGroup,
+                                              mio::osecirvvs::InfectionState, decltype(mio::read_mobility_plain)>(
+                data_dir, params_graph, moving_comp, contact_locations_size, mio::read_mobility_plain, weights);
```
Comment (jubicker) [inline pycode/memilio-simulation/memilio/simulation/osecirvvs.cpp:348]:
```suggestion
                data_dir, params_graph, mobile_comp, contact_locations_size, mio::read_mobility_plain, weights);
```

File: cpp/tests/test_json_serializer.cpp [hunk 5]
```diff
@@ -524,17 +524,17 @@ TEST(TestJsonSerializer, abmTrip)
     EXPECT_EQ(r.value(), trip);
 }
 
-// TEST(TestJsonSerializer, abmWorld) // FIXME: (de)serialize is only partially implemented
```
[the remark this answers is not in the document]
Comment (DavidKerkmann) [reply cpp/tests/test_json_serializer.cpp:527]: It will be used again once the serialise is fixed in #1072 .

File: cpp/models/abm/trip_list.h [hunk 6]
```diff
@@ -51,32 +51,32 @@ struct Trip {
...
      */
-    Trip(PersonId id, TimePoint time_new, LocationId destination, LocationId origin, TransportMode mode_of_transport,
+    Trip(PersonId id, TimePoint time_new, LocationId dest, LocationId org, TransportMode mode_of_transport,
          ActivityType type_of_activity, const std::vector<uint32_t>& input_cells = {})
         : person_id(id)
         , time(mio::abm::TimePoint(time_new.time_since_midnight().seconds()))
-        , migration_destination(destination)
-        , migration_origin(origin)
+        , destination(dest)
+        , origin(org)
```
Comment (mknaranja) [inline cpp/models/abm/trip_list.h:63]: Maybe us **orig** if you want to abbreviate it? Not org?!

File: cpp/simulations/abm_braunschweig.cpp [hunk 7]
```diff
@@ -891,27 +891,27 @@ mio::abm::Simulation create_sampled_simulation(const std::string& input_file, co
...
 
-    set_parameters(world.parameters);
+    set_parameters(model.parameters);
 
-    // Create the world object from statistical data.
-    create_world_from_data(world, input_file, t0, max_num_persons);
-    world.use_migration_rules(false);
+    // Create the model object from statistical data.
+    create_model_from_data(model, input_file, t0, max_num_persons);
+    model.use_mobility_rules(false);
```
Comment (mknaranja) [inline cpp/simulations/abm_braunschweig.cpp:901]: One should maybe deactivate the mobility rules directly before one sets the trips?

File: cpp/simulations/abm_braunschweig.cpp [hunk 8]
```diff
@@ -891,27 +891,27 @@ mio::abm::Simulation create_sampled_simulation(const std::string& input_file, co
...
     auto t_lockdown = mio::abm::TimePoint(0) + mio::abm::days(20);
 
     // During the lockdown, 25% of people work from home and schools are closed for 90% of students.
     // Social events are very rare.
-    mio::abm::set_home_office(t_lockdown, 0.25, world.parameters);
-    mio::abm::set_school_closure(t_lockdown, 0.9, world.parameters);
-    mio::abm::close_social_events(t_lockdown, 0.9, world.parameters);
+    mio::abm::set_home_office(t_lockdown, 0.25, model.parameters);
+    mio::abm::set_school_closure(t_lockdown, 0.9, model.parameters);
+    mio::abm::close_social_events(t_lockdown, 0.9, model.parameters);
```
Comment (mknaranja) [inline cpp/simulations/abm_braunschweig.cpp:912]: Does it make sense to have this here if for Braunschweig one sets much more advanced NPIs later?

File: pycode/memilio-epidata/memilio/epidata/README.rst [hunk 9]
```diff
@@ -213,8 +213,8 @@ DIVI           Germany     county_divi                         ICU, ICU_ventilat
 DIVI           Germany     state_divi                          ICU, ICU_ventilated over time for different states (Bundesländer) with columns ['Date', 'ICU', 'ICU_ventilated', 'ID_State', 'State']
 DIVI           Germany     germany_divi                        ICU, ICU_ventilated over time for whole Germany with columns ['Date', 'ICU', 'ICU_ventilated']
 
-BAA            Germany     migration_bfa_2020_dim401           number of commuters from one county into another indexed by county ids (with eisenach)
-BAA            Germany     migration_bfa_2020_dim400           number of commuters from one county into another indexed by county ids (with eisenach merged into wartburgkreis)
+BAA            Germany     mobility_bfa_2020_dim401           number of commuters from one county into another indexed by county ids (with eisenach)
+BAA            Germany     mobility_bfa_2020_dim400           number of commuters from one county into another indexed by county ids (with eisenach merged into wartburgkreis)
```
Comment (mknaranja) [inline pycode/memilio-epidata/memilio/epidata/README.rst:217]:
```suggestion
BAA            Germany     mobility_bfa_2020_dim401           number of commuters from one county into another indexed by county ids (with county Eisenach)
BAA            Germany     mobility_bfa_2020_dim400           number of commuters from one county into another indexed by county ids (without county Eisenach, i.e., Eisenach merged into county Wartburgkreis)
```

File: pycode/memilio-epidata/memilio/epidata/getCommuterMobility.py [hunk 10]
```diff
@@ -614,8 +616,8 @@ def main():
     arg_dict = gd.cli("commuter_official")
     ref_year = 2022
 
-    abs_tol = 100  # maximum absolute error allowed per county migration
-    rel_tol = 0.01  # maximum relative error allowed per county migration
+    abs_tol = 100  # maximum absolute error allowed per county mobility
+    rel_tol = 0.01  # maximum relative error allowed per county mobility
```
Comment (mknaranja) [inline pycode/memilio-epidata/memilio/epidata/getCommuterMobility.py:620]: Did you make a test run that it returns the same matrix as the one stored in the data folder?

**Review by mknaranja — COMMENTED** [review]
I did a superficial review but would think that it looks very good. Thank you very much for the immense work.

File: pycode/memilio-epidata/memilio/epidata/getCommuterMobility.py (hunk 10 above)
Comment (DavidKerkmann) [reply pycode/memilio-epidata/memilio/epidata/getCommuterMobility.py:620]: Yes, the matrix is the same.

**Review by jubicker — APPROVED** [review]

**Pull request merged**